woocommerce/pinterest-for-woocommerce
# Migrate Setup Task to PHP

### Changes proposed in this Pull Request:



Closes #667 

This PR:
* Removes Onboarding task added using `woocommerce_admin_onboarding_task_list` filter from `@woocommerce/data`. This is because the filter is deprecated.
* Adds the Onboarding task using `TaskLists::add_task`
* Replaces the title of the task to copy provided in the internal discussion here: pb0Spc-3ER-p2. This closes #661.


### Screenshots:
![image](https://user-images.githubusercontent.com/179533/214491315-667fe124-fdae-4967-a035-938dd85f3b77.png)

![image](https://user-images.githubusercontent.com/179533/214492304-b91039b5-6570-4584-a9f5-3dce98f3eac7.png)


### Detailed test instructions:


1. On a fresh WordPress instance, navigate to `Plugins > Add New` and search for "Pinterest for WooCommerce".
2. Install Pinterest for WooCommerce by WooCommerce and activate the plugin.
3. Navigate to WooCommerce > Home and verify that the task now reads "Setup Pinterest for WooCommerce".
4. Right-click on the current page and click on "Inspect Element" to view the browser console.
5. The console should have this warning: `woocommerce_admin_onboarding_task_list is deprecated and will be removed from @woocommerce/data in version 2.10.0. Please use TaskLists::add_task() instead.`
6. Build this PR using `npm run build:zip` and upload the `pinterest-for-woocommerce.zip` file from `Plugins > Add New > Upload Plugin > Install Plugin`. On the next screen, Confirm `Replace current with uploaded`.
7. Navigate to WooCommerce > Home and verify that the task now reads: "Get your products in front of engaged shoppers with Pinterest for WooCommerce".
8. Right-click on the screen and click on "Inspect Element" to view the browser console. Verify that the warning in step 5 is no longer there.
9. Click on the task and complete the Pinterest setup.
10. On completion, navigate to WooCommerce > Home and verify the task "Get your products in front of engaged shoppers with Pinterest for WooCommerce" is checked to indicate task completion.



### Changelog entry

> Update - The Onboarding task to use `TaskLists::add_task` method instead of the deprecated `woocommerce_admin_onboarding_task_list` filter.
